src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -338,20 +338,21 @@ void AresDnsResolver::OnResolvedLocked(void* arg, grpc_error* error) {
     grpc_millis timeout = next_try - ExecCtx::Get()->Now();
     gpr_log(GPR_INFO, "dns resolution failed (will retry): %s",
             grpc_error_string(error));
-    GPR_ASSERT(!r->have_next_resolution_timer_);
-    r->have_next_resolution_timer_ = true;
-    // TODO(roth): We currently deal with this ref manually.  Once the
-    // new closure API is done, find a way to track this ref with the timer
-    // callback as part of the type system.
-    RefCountedPtr<Resolver> self = r->Ref(DEBUG_LOCATION, "retry-timer");
-    self.release();
-    if (timeout > 0) {
-      gpr_log(GPR_DEBUG, "retrying in %" PRIdPTR " milliseconds", timeout);
-    } else {
-      gpr_log(GPR_DEBUG, "retrying immediately");
+    if (!r->have_next_resolution_timer_) {
+      // TODO(roth): We currently deal with this ref manually.  Once the
+      // new closure API is done, find a way to track this ref with the timer
+      // callback as part of the type system.
+      RefCountedPtr<Resolver> self = r->Ref(DEBUG_LOCATION, "retry-timer");
+      self.release();
+      if (timeout > 0) {
+        gpr_log(GPR_DEBUG, "retrying in %" PRIdPTR " milliseconds", timeout);
```

Comment:
Please add a log message for the case where we are not scheduling a timer due to one already being pending.

Same thing in both implementations.

Reply:
Done.